runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -1017,21 +1294,229 @@ MM_SchedulingDelegate::calculateEdenSize(MM_EnvironmentVLHGC *env)
 	Trc_MM_SchedulingDelegate_calculateEdenSize_Exit(env->getLanguageVMThread(), (_edenRegionCount * regionSize));
 }
 
-UDATA
+void
+MM_SchedulingDelegate::moveTowardRecommendedEden(MM_EnvironmentVLHGC *env, double edenChangeSpeed) 
+{
+	Assert_MM_true(edenChangeSpeed <= 1 && edenChangeSpeed >= 0);
+
+	if ((0 == _historicalPartialGCTime) || (0 == _averagePgcInterval)) {
+		/* Until we have collected any information about PGC time, we don't have the data we need to make informed decision about eden size  */
+		return;
+	}
+
+	uintptr_t recommendedEdenSizeBytes = calculateRecommendedEdenSize(env);
+
+	uintptr_t currentIdealEdenBytes = getIdealEdenSizeInBytes(env);
+	uintptr_t currentIdealEdenRegions = _idealEdenRegionCount;
+
+	/* 
+	 * The closer edenChangeSpeed is to 1, the larger the move towards recommendedEdenSizeBytes will be. 
+	 * 1 implies that eden should move all the way towards recommendedEdenSizeBytes.
+	 */
+	intptr_t edenChange = (intptr_t)recommendedEdenSizeBytes - currentIdealEdenBytes;
+	intptr_t targetEdenChange = (intptr_t)(edenChange * edenChangeSpeed);
+	uintptr_t targetEdenBytes = currentIdealEdenBytes + targetEdenChange;
+	uintptr_t targetEdenRegions = targetEdenBytes / _regionManager->getRegionSize();
+
+	_edenSizeFactor = (intptr_t)targetEdenRegions - currentIdealEdenRegions;
+}
+
+void
+MM_SchedulingDelegate::checkEdenSizeAfterPgc(MM_EnvironmentVLHGC *env, bool globalSweepHappened)
+{
+	if (!_extensions->statupPhaseFinished) {
```

Comment:
What was your overall impression, how useful/important is to consider this case? (While fun learning exercise), for sake of simplicity, I'm willing to remove this.

Reply:
From what I've seen in testing, in startup, GC times/overhead as well as survival rate (and a few other factors) are not representative of the actual regular allocation profile of the application (like at all...). That being said, making decisions on eden size during this time, might cause eden to shrink when it should expand, or vise versa.